common/Util.cpp

```diff
@@ -16,6 +16,11 @@
 
 #include <csignal>
 #include <poll.h>
+
+#ifdef HAVE_SYS_RANDOM_H
+#  include <sys/random.h>
+#endif
+
```

Comment:
But `HAVE_SYS_RANDOM_H` is always disabled (even on my linux system). When I run `make` I can see that my system finds `sys/random.h` header but here `sys/random.h` never includes because `HAVE_SYS_RANDOM_H` is defined in:
- `config_unused.h`       as `#define HAVE_SYS_RANDOM_H 1`
- `config_unused.h.in`   as `#undef HAVE_SYS_RANDOM_H`
and those two files are ignored in .gitignore so you can't see HAVE_SYS_RANDOM_H anywhere.

I don't know why `AC_CHECK_HEADERS([sys/random.h]` define `HAVE_SYS_RANDOM_H` in files that ignored, instead of defining them in e.g. `config.h.in` (or some other files that visible and not ignored by .gitignore).
Hi @Ashod; is there any solution for this or am I doing something wrong here? Thanks.

Reply:
I imagine the better approach is not just checking for the header, but to actual check that the function exists and compiles. The way to do that is to have a minimal code that must compile for the feature to be enabled. We do this for other features. For example, see how it's done to check for C++17. Something similar should work in this case, with the result of having some compile-time directive set. The code can then have `#ifdef` based on that directive to enable this code vs the existing one.

Hope this helps.